### ISSUE TITLE
[file-extension-in-import] Does not work properly in TypeScript projects with allowImportingTsExtensions

### ISSUE THREAD

**Issue by Zamiell** [eslint-community/eslint-plugin-n#134]
### Environment

Hello,

With the following config:

```
"n/file-extension-in-import": ["error", "always"],
```

Will autofix this:

```ts
import { foo } from "./foo";
```

To this:

```ts
import { foo } from "./foo.js";
```

However, this is a bug in TypeScript projects that use `allowImportingTsExtensions`. In such projects, having a ".js" import is a bug, because the assumption is that our entire application is written in TypeScript, and no JavaScript files will probably ever exist.

Proposal:

Instead of fixing to `.js`, the rule should auto-fix to `.ts` when the "tsconfig.json" has the `allowImportingTsExtensions` option set to true.

### What rule do you want to report?

file-extension-in-import

### Participation

- [ ] I am willing to submit a pull request for this issue.

**Comment by scagood** [comment]
🤔 This makes sense, I will try to look next week. Thanks for the report!

**Comment by Zamiell** [comment]
@scagood I see that you've fixed the bug on your fork, good work! Anything I can do to move this issue along?

**Comment by scagood** [comment]
There are two things holding this back:
1. I need to make some fixes (in #139) specifically it looks like I may have broken the `file-extension-in-import` rule.
2. We want to wait for eslint v9 to be released so we're sure that the plugin is stable with that release before we introduce some possible breaking changes.

I should get some time on Thursday to look into the broken rule problem.
